### DISCUSSION TITLE
Problem with data storage from timelineVariable

### DISCUSSION THREAD

**Discussion by moramaldonado** [jspsych/jsPsych#752] [Q&A]
Hi all, 

I am currently running an experiment where I have (roughly) the following timeline structure: 

```js
var trials = {
  type: 'html-keyboard-response',
  stimulus: jsPsych.timelineVariable('sentence'),
  data: {sentence: jsPsych.timelineVariable('sentence')} 
}

var procedure = {
  timeline: [trials],
  timeline_variables: create_stimuli()
}
```

where create_stimuli is a js function that returns me a list of dictionaries, each with an entry "sentence".

After running around 50 participants, I noticed that the "sentence" value that I am storing in the data and the "sentence" value that I am giving as an argument for the plugin is not always the same: very rarely it just picks up two different values from my list of dictionaries. 

I have spent some time trying to figuring out if there is some issue with my code, but it seems to be no only rare but also aleatory, what makes me think that it's related to some refreshing/connectivity issue. 

Do you happen to know why this might be happening? Any idea is very belcome!

Thanks in advance

Mora

**Comment by jodeleeuw** [comment]
That is extremely odd. I don't see any obvious reason why this should happen based on the timeline structure you provided. `create_stimuli()` should run during the setup phase of the experiment, so there's nothing unusual about this example.

Is it possible to share a link to the live experiment? You could privately send it to me if you don't want it made public right now.

**Reply by moramaldonado** [reply]
Thanks for your answer! I am bit unsure of how to send private messages here... but well, it doesn't really matter. 

Here is a link of the live experiment: http://mmaldonadosyncog.ppls.ed.ac.uk/experiment/neg/try.html

It's a reduced version I was using to test this issue (it doesn't contain the complete experiment), but it does have the problem I mentioned (like, of the 4 times I ran it myself there were a couple of items with this problem, but not always the same items and not every time I did it..).

**Reply by jodeleeuw** [reply]
Is it one of the timelines in the timeline.js file in particular? Or any of them?

**Reply by moramaldonado** [reply]
I noticed it for the sentence_production and negation_production ones.

**Reply by moramaldonado** [reply]
Now that I have gone through the whole data in order, I think what might be happening is that in some cases the output of the plugin is not properly stored but rather copied from the one in the trial before.

**Reply by jodeleeuw** [reply]
Hi @moramaldonado sorry that I didn't follow up earlier. Did you get this resolved?
